dipy/reconst/dti.py

```diff
@@ -1242,22 +1283,41 @@ def wrapped_fit_tensor(design_matrix, data, step=step,
             size = np.prod(shape)
             step = int(step) or size
             if step >= size:
-                return fit_tensor(design_matrix, data, *args, **kwargs)
+                if return_S0_hat:
```

Comment:
You're right, this can be simplified. Though not anywhere else in this function as far as I can see.